staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go

```diff
@@ -35,18 +36,20 @@ var (
 )
 
 // PanicHandlers is a list of functions which will be invoked when a panic happens.
-var PanicHandlers = []func(interface{}){logPanic}
+var PanicHandlers = []func(context.Context, interface{}){logPanic}
 
 // HandleCrash simply catches a crash and logs an error. Meant to be called via
 // defer.  Additional context-specific handlers can be provided, and will be
 // called in case of panic.  HandleCrash actually crashes, after calling the
 // handlers and logging the panic message.
 //
 // E.g., you can provide one or more additional handlers for something like shutting down go routines gracefully.
+//
+// TODO(pohly): logcheck:context // HandleCrashWithContext should be used instead of HandleCrash in code which supports contextual logging.
 func HandleCrash(additionalHandlers ...func(interface{})) {
```

Comment:
deprecate?

Reply:
Can we express HandleCrash through HandleCrashWithContext?

Reply:
> deprecate?

https://grep.app/search?q=utilruntime.HandleCrash shows that `HandleCrash` is used quite a bit outside of Kubernetes. Marking it as deprecated IMHO is too strong: it implies that we expect all of those downstream consumers of `utilruntime` to  change their code although it works perfectly fine for them.

It's only code which cares about contextual logging which should use `HandleErrorWithContext`.

> Can we express HandleCrash through HandleCrashWithContext?

That would interfere with stack unwinding in callbacks, for example `logError`: as it stands now, they know that they need to skip HandleCrash or HandleCrashWithContext. If HandleCrash calls HandleCrashWithContext, that adds another level and the callback won't know that.